test/src/test/java/org/corfudb/integration/CorfuReplicationClusterConfigIT.java

```diff
@@ -1494,14 +1499,13 @@ public void testActiveLockRelease() throws Exception {
             txnContext.commit();
         }
 
-        log.info("Active's lock is released!");
-        TimeUnit.SECONDS.sleep(lockInterval);
+        Assert.assertEquals(0, activeLockTable.count());
+        log.info("Active's lock table cleared!");
 
-        // Release Active's lock again
-        try (TxnContext txnContext = activeCorfuStore.txn(CORFU_SYSTEM_NAMESPACE)) {
-            txnContext.clear(activeLockTable);
-            txnContext.commit();
-        }
+        // The detection of lock release will be finished in 1 second.  However, replication stops after this event is
+        // asynchronously processed by the DiscoveryService.  So add a sleep of 5 seconds accounting for the processing
+        // delay
+        TimeUnit.SECONDS.sleep(lockReleaseProcessingDelay);
```

Comment:
while this might work - won't 5 seconds still end up being slow in systems where the git actions is running slow?

Reply:
Once the lock table is deleted, the missing lock is identified by a thread whose cadence we reduced to 1 second.

On detecting the absence of the lock, an event is generated to stop replication.  Since this is an IT, we cannot identify when the event was processed.  So basing the wait period on a reasonable assumption that after the detection(1 second delay max), the event will get processed within the next 4 seconds.

Reply:
no longer using the sleep mechanism as we did identify a deterministic way to check that lock release was processed.